flax/core/scope.py

```diff
@@ -205,21 +206,21 @@ def wrapper(*args, **kwargs):
       return fn(scope.rewound(), *args, **kwargs)
     return wrapper
 
-  def get_kind(self, kind: str, mutable: bool = False) -> MaybeFrozenKind:
-    """Returns all variable of a given kind."""
-    if kind not in self._variables:
+  def collection(self, col: str, mutable: bool = False) -> MaybeFrozenCollection:
```

Comment:
The return value type here is also much clearer -- before this function returned a "kind"?